pkg/fsp/header.go

```diff
@@ -0,0 +1,206 @@
+// Copyright 2018 the LinuxBoot Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package fsp implements FSP info header parsing
+package fsp
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/u-root/u-root/pkg/uio"
+)
+
+// TODO support FSP versions < 2.0
+// TODO implement FSP_INFO_EXTENDED_HEADER
+
+// FSP 2.0 specification
+// https://www.intel.com/content/dam/www/public/us/en/documents/technical-specifications/fsp-architecture-spec-v2.pdf
+
+// values from the FSP 2.0 spec
+var (
+	Signature = [4]byte{'F', 'S', 'P', 'H'}
+)
+
+// constants from the FSP 2.0 spec
+const (
+	// size of the shared part of the header across FSP spec revisions
+	CommonHeaderLength = 12
+	HeaderV3Length     = 72
+	// FSP 2.0
+	CurrentSpecVersion = SpecVersion(0x20)
+	HeaderV3Revision   = 3
+)
+
+// InfoHeaderRev3 represents the FSP_INFO_HEADER structure revision 3 (FSP
+// 2.0) as defined by Intel.
+type InfoHeaderRev3 struct {
```

Comment:
Would prefer a stringer on this struct rather than have the prints in the cmd

Reply:
will do! Thanks for the review :)

Reply:
one question before I do that: my understanding of the stringers in Go is that it's used to describe a type with a string. While I believe there's no recommendation about this, I don't remember multi-line stringers, rather short representations of the object.

The prints in cmds/fspinfo are a verbose, multi-line representation. Instead of `String()`, should we have instead another method (e.g. `Summary()`)?

Reply:
Honestly, I'm ok with either. I'll leave it up to you.

Reply:
ok, I'll go with `Summary` (we can change our mind later anyway)

Reply:
updated